pkg/admission/reservedmetadata/admission.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reservedmetadata
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/utils/strings/slices"
+)
+
+const (
+	PluginName = "apis.kcp.dev/ReservedMetadata"
+)
+
+var (
+	annotationAllowList = []string{
+		"scheduling.kcp.dev/placement",
+		"internal.scheduling.kcp.dev/negotiation-workspace",
+	}
+
+	labelAllowList = []string{
+		"experimental.workload.kcp.dev/scheduling-disabled",
+		"state.internal.workload.kcp.dev/*",
```

Comment:
are these necessary for e2e to work? Internal APIs should not show up in these lists.

Reply:
Agreed, however this has been marked as mutable by coordination controller in the sheet. I can remove and see if e2e passes (pretty sure it will as everything runs as system:masters).

Reply:
oh this then means that we have to change the name to non-internal.

Reply:
and it needs a story how we authorize its change.

Reply:
as discussed, we just left `experimental.workload.kcp.dev/scheduling-disabled` in the allow list of labels, removed everything else.